src/ui/layout/component.tsx

```diff
@@ -0,0 +1,89 @@
+import classNames from "classnames";
+import React, { Component, ReactElement } from "react";
+import {
+  AlignContent,
+  AlignItems,
+  Display,
+  FlexDirection,
+  FlexWrap,
+  JustifyContent,
+  Position,
+} from "./models";
+import "./style.scss";
+
+interface PublicProps {
+  className?: string;
+  padding?: Padding;
+  margin?: Margin;
+  display?: Display;
+  flexDirection?: FlexDirection;
+  flexWrap?: FlexWrap;
+  justifyContent?: JustifyContent;
+  alignItems?: AlignItems;
+  alignContent?: AlignContent;
+  position?: Position;
+}
+
+interface Spacing {
+  y?: size;
+  x?: size;
+  t?: size;
+  b?: size;
+  r?: size;
+  l?: size;
+  [key: string]: size | undefined;
+}
+
+type size = 1 | 2 | 3 | 4 | 5;
```

Comment:
nit: is the lowercase `s` intentional? wondering since the rest of the types are all capitalized